tests/system/common.py

```diff
@@ -763,3 +763,11 @@ def task_by_name(tasks, name):
             return task
 
     assert False, "Did not find task with name %s in this list of tasks: %s" % (name, tasks,)
+
+
+@retrying.retry(wait_fixed=100, stop_max_attempt_number=30, retry_on_exception=ignore_exception, stop_max_delay=30000)
+def assert_event(event_type, events):
+    """ Skip events until event with given type was found."""
```

Comment:
btw. this can be potentially dangerous. If the order of the events we are expecting is not guaranteed...

if we do
assert_event(a)
assert_event(b)

but the events arrive as:
b
a

we will never get success out of this...

Reply:
That is true but I think for the events we check the ordering is guaranteed.